lib/private/handlers/container-images.ts

```diff
@@ -105,7 +105,10 @@ export class ContainerImageAssetHandler implements IAssetHandler {
 
     const destination = await replaceAwsPlaceholders(this.asset.destination, this.host.aws);
     const ecr = await this.host.aws.ecrClient({
-      ...destination,
+      assumeRoleArn: destination.assumeRoleArn,
```

Comment:
Explicitly build `ClientOptions` from `AwsDestination`. The fact they are structurally compatible is coincidental. This also enforces better type checking that `cdk-assets` depends on the appropriate version of `@aws-cdk/cloud-assembly-schema`.

Reply:
I agree. This comment should be in the code, so that future maintainers also know this fact and don't accidentally undo the explicit copying.

Reply:
Better yet -- make it a function, and put that comment on the function description, then you can call it from multiple places (rather than linking a GitHub comment from multiple other GitHub comments).